src/main/java/com/example/Storehouse/service/ResponseService.java

```diff
@@ -0,0 +1,170 @@
+package com.example.Storehouse.service;
+
+import com.example.Storehouse.entity.Products;
+import com.example.Storehouse.entity.RefundProducts;
+import com.example.Storehouse.exception.exceptions.WrongQuantityException;
+import com.example.Storehouse.logger.Print;
+import com.example.Storehouse.models.product.Product;
+import com.example.Storehouse.models.product.RequiredProduct;
+import com.example.Storehouse.models.request.CheckQuantityOfProductRequest;
+import com.example.Storehouse.models.request.MultiProductRequest;
+import com.example.Storehouse.models.request.RefundRequest;
+import com.example.Storehouse.models.request.SingleProductRequest;
+import com.example.Storehouse.models.response.CheckQuantityResponse;
+import com.example.Storehouse.models.response.MultiProductsResponse;
+import com.example.Storehouse.models.response.RefundResponse;
+import com.example.Storehouse.models.response.SingleProductResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class ResponseService implements iResponseService {
+    @Autowired
+    private Print print;
+    @Autowired
+    private  StorehouseService service;
+    private static final List<Product> productList = new ArrayList<>();
```

Comment:
bad practice to have static variable for REAL objects, this state can be shared between multiple objects. and can potentially rewrite data for another client. all models should be created only for each client request and removed when request done his job. do not save state in static variable.

Reply:
done